rx-preferences/src/main/java/com/f2prateek/rx/preferences2/RealPreference.java

```diff
@@ -15,8 +13,12 @@
 final class RealPreference<T> implements Preference<T> {
   /** Stores and retrieves instances of {@code T} in {@link SharedPreferences}. */
   interface Adapter<T> {
-    /** Retrieve the value for {@code key} from {@code preferences}. */
-    T get(@NonNull String key, @NonNull SharedPreferences preferences);
+    /**
+     * Retrieve the value for {@code key} from {@code preferences}, or {@code defaultValue}
+     * if the preference is unset, or was set to {@code null}.
+     */
+    @NonNull T get(@NonNull String key, @NonNull SharedPreferences preferences,
+        @NonNull T defaultValue);
```

Comment:
Only confirming: while technically an ABI change, this class is package private and thus this change is fine. Consumers of this library implement `Preference.Converter` and not the `Adapter` interface.

Reply:
Should have read your comments first, you said the same :-)

Reply:
More is more.